File: typescipt/src/language/expressions/functions/mapping.ts


export class Mapping {
   public string VariableName
   public VariableType VariableType
   public VariableSource VariableSource

   constructor(variableName: string, variableType: VariableType, variableSource: VariableSource) {
     VariableName = variableName;
     VariableType = variableType;
     VariableSource = variableSource;
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/Mapping.cs
using Lexy.Compiler.Language.VariableTypes;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class Mapping
{
    public string VariableName { get; }
    public VariableType VariableType { get; }
    public VariableSource VariableSource { get; }

    public Mapping(string variableName, VariableType variableType, VariableSource variableSource)
    {
        VariableName = variableName;
        VariableType = variableType;
        VariableSource = variableSource;
    }

    public VariableUsage ToUsedVariable(VariableAccess access)
    {
        var variablePath = IdentifierPath.Parse(VariableName);
        return new VariableUsage(variablePath, null, VariableType, VariableSource, access);
    }
}